### ISSUE TITLE
WhiteSpaceBear: Remove trailing whitespaces and trailing blank lines

### ISSUE THREAD

**Issue by meetmangukiya** [coala/coala-bears#1793]
I think removing trailing whitespace exists in some bears, but is language dependent. Haven't found a bear that removes trailing blank lines though

**Comment by jayvdb** [comment]
This would go into SpaceConsistencyBear I think?

**Comment by meetmangukiya** [comment]
@jayvdb yes, you are right, it should go into SpaceConsistencyBear :+1:

**Comment by khanchi97** [comment]
@danlls are you still working on this?If not, then I would like to do it.

**Comment by danlls** [comment]
I'm not, you can go ahead. Somebody can withdraw me from assignee

**Comment by khanchi97** [comment]
@jayvdb @meetmangukiya I want to ask one thing.Let say in a file, if there are some trailing blanklines at EOF with whitespaces in these lines.So now when running `SpaceConsistencyBear`(I have made changes for `allow_trailing_blanklines` option), first it offers to delete those lines using `applypatch` action and if I use this action and after deleting those lines, now coala offers to remove `whitespace` from those lines which are already deleted by `allow_trailing_blanklines` option.

**Comment by Makman2** [comment]
Pick the behaviour that is most intuitive and most understandable from user perspective ;)

**Comment by khanchi97** [comment]
No, I want to ask how I can fix this problem that if those lines have already deleted by `allow_trailing_blanklines` option then why in next step coala offers to remove whitespace from those lines which have been already deleted.

**Comment by Makman2** [comment]
Fix that behaviour and make the bear not emit patches for those lines any more :+1:
